### PR TITLE
Paraswap: Fee redistributor updates

### PR DESCRIPTION
**Pull request by facuspagnuolo**
This PR includes several changes to the actions deployed for Paraswap's fee redistributor smart vault

### Mainnet
- [x] New `PermissionsManager` in control of the entire architecture
- [x] Upgrade `ERC20Claimer` to support configuring the token out
- [x] New `MetamaskClaimer` action
- [x] Disable `SwapFeeSetter` action
- [x] Reset swap fees
- [x] Turn on withdraw fees

### Polygon, Arbitrum, Optimism, BNB
- [x] New `PermissionsManager` in control of the entire architecture
- [x] Upgrade `ERC20Claimer` to support configuring the token out
- [x] New `MetamaskClaimer` action
- [x] New `ConnextBridger` action
- [x] Disable `Withdrawer` action
- [x] Disable `SwapFeeSetter` action
- [x] Reset swap fees

### Fantom
- [x] New `PermissionsManager` in control of the entire architecture
- [x] Upgrade `ERC20Claimer` to support configuring the token out
- [x] New `FantomBridger` action
- [x] Disable `Withdrawer` action
- [x] Disable `SwapFeeSetter` action
- [x] Reset swap fees

### Avalanche
- [x] New `PermissionsManager` in control of the entire architecture
- [x] Upgrade `ERC20Claimer` to support configuring the token out
- [x] New `WormholeBridger` action
- [x] Disable `Withdrawer` action
- [x] Disable `SwapFeeSetter` action
- [x] Reset swap fees

### CODE REVIEW

File: tasks/2023062100-paraswap-fee-redistributor-updates-mainnet/input.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+const ParaswapFeeRedistributorWithdrawer = new Task('2023031300-paraswap-withdrawer')
+
+/* eslint-disable no-secrets/no-secrets */
+
+export default {
+  namespace: 'mimic-v2.paraswap-fee-redistributor',
+  version: 'v1',
+  from: DEPLOYER_2,
+  owner: '0x619bbf92fd6ba59893327676b2685a3762a49a33',
+  mimic: TESTING_EOA,
```
Comment (dmf7z) [inline tasks/2023062100-paraswap-fee-redistributor-updates-mainnet/input.ts:19]: shouldn't we set directly the new admin? 
`0x26A0c29261d18C49e6Cf0Cee175018274D58f90C`
please confirm @brunitob

**Review by dmf7z — CHANGES_REQUESTED** [review]

File: tasks/2023062101-paraswap-fee-redistributor-updates-polygon/index.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        target: nativeClaimer,
+        changes: [
+          { grant: false, permission: { who: owner, what: 'authorize' } },
+          { grant: false, permission: { who: owner, what: 'unauthorize' } },
+          { grant: false, permission: { who: mimic, what: 'authorize' } },
+          { grant: false, permission: { who: mimic, what: 'unauthorize' } },
+        ],
+      },
+      {
+        target: withdrawer,
```
Comment (dmf7z) [inline tasks/2023062101-paraswap-fee-redistributor-updates-polygon/index.ts:137]: we should deprecate withdraw tasks on all networks but mainnet, right? (eg. remove the withdraw permission)

File: tasks/2023062100-paraswap-fee-redistributor-updates-mainnet/index.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        thresholdAmount: input.metamaskClaimer.thresholdAmount,
+        gasPriceLimit: input.metamaskClaimer.gasPriceLimit,
+        gasToken: input.metamaskClaimer.gasToken,
+        relayer,
+      },
+    ],
+    { ...create3Params }
+  )
+
+  await executePermissionChanges(
```
Comment (dmf7z) [inline tasks/2023062100-paraswap-fee-redistributor-updates-mainnet/index.ts:67]: we should remove the SwapFeeSetter from all networks

**Review by dmf7z — CHANGES_REQUESTED** [review]

File: tasks/2023062100-paraswap-fee-redistributor-updates-mainnet/index.ts (hunk 3 above)
Comment (dmf7z) [reply tasks/2023062100-paraswap-fee-redistributor-updates-mainnet/index.ts:67]: Afterwards will need to configure the final fee with cap on the withdraw task on Mainnetm, right?

File: constants/tokens/bsc.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable no-secrets/no-secrets */
 
 export const WBNB = '0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c'
-export const WETH = '0x2170ed0880ac9a755fd29b2688956bd959f933f8'
+export const WETH = '0x2170Ed0880ac9A755fd29B2688956BD959F933F8'
```
[the remark this answers is not in the document]
Comment (facuspagnuolo) [reply constants/tokens/bsc.ts:4]: I don't think so, `0x2170ed0880ac9a755fd29b2688956bd959f933f8` is correct

File: tasks/2023062100-paraswap-fee-redistributor-updates-mainnet/index.ts (hunk 3 above)
Comment (facuspagnuolo) [reply tasks/2023062100-paraswap-fee-redistributor-updates-mainnet/index.ts:67]: which is the final fee we should set then?

**Comment by facuspagnuolo** [comment]
@dmf7z please take another look, I've added a few changes to disable the withdrawer action on all L2s and disable the swap fee setter action completely. The part missing is the withdraw fee for mainnet.

File: constants/tokens/bsc.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (brunitob) [reply constants/tokens/bsc.ts:4]: Confirmed 100% sure it's `0x2170ed0880ac9a755fd29b2688956bd959f933f8` it's the official [Binance-Peg Ethereum Token (ETH)](https://bscscan.com/token/0x2170ed0880ac9a755fd29b2688956bd959f933f8)

There's no WETH, only Ethereum Token on BSC, but it's a BEP20

[the remark this answers is not in the document]
Comment (brunitob) [reply constants/tokens/bsc.ts:4]: Confirmed 100% sure it's `0x2170ed0880ac9a755fd29b2688956bd959f933f8` it's the official [Binance-Peg Ethereum Token (ETH)](https://bscscan.com/token/0x2170ed0880ac9a755fd29b2688956bd959f933f8)

There's no WETH, only Ethereum Token on BSC, but it's a BEP20

File: tasks/2023062100-paraswap-fee-redistributor-updates-mainnet/index.ts (hunk 3 above)
Comment (brunitob) [reply tasks/2023062100-paraswap-fee-redistributor-updates-mainnet/index.ts:68]: [Mimic will charge a 2% fee for the assets processed through the Smart Vaults with a 5K USD monthly cap.](https://gov.paraswap.network/t/psp-ip-23-implementing-mimic-smart-vaults-for-fee-collection-and-conversion/1232#h-3-fees-10)

File: tasks/2023062100-paraswap-fee-redistributor-updates-mainnet/input.ts (hunk 1 above)
Comment (brunitob) [reply tasks/2023062100-paraswap-fee-redistributor-updates-mainnet/input.ts:19]: Yes, 
`0x26A0c29261d18C49e6Cf0Cee175018274D58f90C` eth, polygon, arb, opt, avax,bsc, 
`0x2D956351fA705a5FCd5fE0e99363Ef9CcCc84921` fantom

@dmf7z lets do ASAP the confirm&migration task.

File: tasks/2023062106-paraswap-fee-redistributor-updates-avalanche/input.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  NativeClaimer: ParaswapFeeRedistributor,
+  ERC20Claimer: ParaswapFeeRedistributor,
+  SwapFeeSetter: ParaswapFeeRedistributor,
+  Withdrawer: ParaswapFeeRedistributorWithdrawer,
+  erc20Claimer: {
+    tokenOut: tokens.WETH,
+  },
+  wormholeBridger: {
+    destinationChainId: 1,
+    allowedTokens: [tokens.WETH],
```
Comment (brunitob) [inline tasks/2023062106-paraswap-fee-redistributor-updates-avalanche/input.ts:33]: This is ok? Wormhole/CCTP only supports USDC

File: tasks/2023062106-paraswap-fee-redistributor-updates-avalanche/input.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    maxRelayerFeePct: fp(0.05), // 5%
```
Comment (brunitob) [inline tasks/2023062106-paraswap-fee-redistributor-updates-avalanche/input.ts:34]: Relayer Fee is always 35 USD, 
5% -> 35 USDC -> min amount to bridge it's 700 USDC
We should set the `maxRelayerFeePct: fp(0.07),`

Comment (brunitob) [reply tasks/2023062106-paraswap-fee-redistributor-updates-avalanche/input.ts:34]:
```suggestion
    maxRelayerFeePct: fp(0.07), // 7%
```

**Review by dmf7z — APPROVED** [review]

**Review by brunitob — APPROVED** [review]
Go Go Go

**Pull request merged**